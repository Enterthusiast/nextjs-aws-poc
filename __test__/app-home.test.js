/* eslint-env jest */

import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import AppHome from '../components/app-home.js'

describe('With Enzyme', () => {
  it('AppHome shows subtitle "A serverless website."', () => {
    const app = mount(<AppHome />)

    expect(app.find('.subtitle').text()).toEqual('A serverless website.')
  })
})

describe('With Snapshot Testing', () => {
  it('AppHome match snapshot', () => {
    const component = renderer.create(<AppHome />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})