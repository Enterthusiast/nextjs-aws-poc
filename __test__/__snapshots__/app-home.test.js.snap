// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing AppHome match snapshot 1`] = `
<div
  className="sc-bdVaJa jMztiP"
>
  <div>
    <section
      className="section"
    >
      <div
        className="container"
      >
        <h1
          className="title sc-bwzfXH gLzMBj"
        >
          Enterthusiast code playground
        </h1>
        <p
          className="subtitle"
        >
          A 
          <strong>
            serverless
          </strong>
           website.
        </p>
      </div>
    </section>
    <section
      className="section is-medium"
    >
      <div
        className="container"
      >
        <p>
          Hello interweb wanderer \\o/
          <br />
          <br />
          Here I'm playing with next.js and aws lambda with a packager and uploader named scandium.
          <br />
          I also added a list of article using dynamoDB with aws lambda again.
          <br />
          <a
            href="/default/article"
            onClick={[Function]}
          >
            Article list
          </a>
          <br />
          <br />
          By the way we are also connected to the blizzard api, feel free to check your own character.
          <br />
          <a
            href="/default/wow-character-finder"
            onClick={[Function]}
          >
            WoW character finder
          </a>
        </p>
      </div>
    </section>
  </div>
</div>
`;
